### ISSUE TITLE
dotenv should look up in parent directories for .env

### ISSUE THREAD

**Issue by cdignam-segment** [joho/godotenv#165]
My `.env` file is at the root of my project, but when running tests, the current working directory is within test modules.

This would match the behavior of https://github.com/dotenv-rs/dotenv/tree/3c1a77bc95821777e5ceb996c5e0b082f2a3ea38#usage and https://github.com/theskumar/python-dotenv.

**Comment by joho** [comment]
Yep, you're right. Will bring that feature in as a breaking change

**Comment by cdignam-segment** [comment]
@joho I made a PR to resolve this issue: https://github.com/joho/godotenv/pull/166

It would be great to get your feedback!

**Comment by wengtad** [comment]
Hi, any updates?

**Comment by joho** [comment]
I've been umming and ahhing on this one. I know I was very supportive when originally opened, but ever since I've been worrying about directory traversal attacks.

I'm about to close the door on breaking changes (see #182) but will hopefully include this one before I do (if I can come to comfort on the traversal risks)

**Comment by rcy** [comment]
@joho what's your comfort level at around this now?  I'm running into the same issue here with wanting my `$PROJECT_ROOT/.env.test` to be loaded when running go tests.  Any suggested workarounds would be appreciated as well!

**Comment by kirici** [comment]
@rcy 
> Any suggested workarounds would be appreciated as well!

[direnv](https://github.com/direnv/direnv/), if set, checks and loads .env/.envrc in parent directories as well, making them available to your shell. That comes with the added benefit of not having to load/source the same file in every application/script.
